programs/drift/src/instructions/user.rs

```diff
@@ -1013,6 +1017,479 @@ pub fn handle_transfer_deposit<'c: 'info, 'info>(
+            borrow_to_oracle_price_data.price,
+            borrow_to_spot_market.get_precision().cast()?,
+        )?;
+
+        controller::spot_position::update_spot_balances_and_cumulative_deposits_with_limits(
+            borrow_amount as u128,
+            &SpotBalanceType::Borrow,
+            &mut borrow_to_spot_market,
+            to_user,
+        )?;
+
+        let deposit_record_id = get_then_update_id!(borrow_to_spot_market, next_deposit_record_id);
+        let deposit_record = DepositRecord {
+            ts: clock.unix_timestamp,
+            deposit_record_id,
+            user_authority: *authority_key,
+            user: to_user_key,
+            direction: DepositDirection::Withdraw,
+            amount: borrow_amount,
+            oracle_price: borrow_to_oracle_price_data.price,
+            market_index: borrow_to_market_index,
+            market_deposit_balance: borrow_to_spot_market.deposit_balance,
+            market_withdraw_balance: borrow_to_spot_market.borrow_balance,
+            market_cumulative_deposit_interest: borrow_to_spot_market.cumulative_deposit_interest,
+            market_cumulative_borrow_interest: borrow_to_spot_market.cumulative_borrow_interest,
+            total_deposits_after: to_user.total_deposits,
+            total_withdraws_after: to_user.total_withdraws,
+            explanation: DepositExplanation::Transfer,
+            transfer_user: Some(from_user_key),
+        };
+        emit!(deposit_record);
+    }
+
+    drop(deposit_from_spot_market);
+    drop(deposit_to_spot_market);
+    drop(borrow_from_spot_market);
+    drop(borrow_to_spot_market);
+
+    from_user.meets_withdraw_margin_requirement_and_increment_fuel_bonus_swap(
+        &perp_market_map,
+        &spot_market_map,
+        &mut oracle_map,
+        MarginRequirementType::Initial,
+        deposit_from_market_index,
+        deposit_transfer.cast::<i128>()?,
+        borrow_from_market_index,
+        -borrow_amount.cast::<i128>()?,
+        user_stats,
+        clock.unix_timestamp,
+    )?;
+
+    to_user.meets_withdraw_margin_requirement_and_increment_fuel_bonus_swap(
+        &perp_market_map,
+        &spot_market_map,
+        &mut oracle_map,
+        MarginRequirementType::Initial,
+        deposit_to_market_index,
+        -deposit_transfer.cast::<i128>()?,
+        borrow_to_market_index,
+        borrow_amount.cast::<i128>()?,
+        user_stats,
+        clock.unix_timestamp,
+    )?;
+
+    validate_spot_margin_trading(
+        from_user,
+        &perp_market_map,
+        &spot_market_map,
+        &mut oracle_map,
+    )?;
+
+    from_user.update_last_active_slot(slot);
+
+    validate_spot_margin_trading(to_user, &perp_market_map, &spot_market_map, &mut oracle_map)?;
+
+    to_user.update_last_active_slot(slot);
+
+    if from_user.is_being_liquidated() {
+        from_user.exit_liquidation();
+    }
+
+    if to_user.is_being_liquidated() {
+        to_user.exit_liquidation();
+    }
```

Comment:
is this safe? or should you use `validate_user_not_being_liquidated` here to check if the user is out of liquidation?

Reply:
they clear margin check above so you know they arent being liquidated